modules/es6.promise.js

```diff
@@ -107,7 +108,27 @@ function $resolve(value){
 }
 
 // constructor polyfill
-if(!(isFunction(P) && isFunction(P.resolve) && P.resolve(test = new P(function(){})) == test)){
+var workingPromise = isFunction(P) && isFunction(P.resolve) &&
+  P.resolve(test = new P(function(){})) == test;
+// Firefox ~33 had broken subclass support, test that.
+if (workingPromise){
+  try { // protect against bad/buggy Object.setPrototype
+    var P2 = function(x) {
+      var self = new P(x);
+      setProto.set(self, P2.prototype);
+      return self;
+    };
+    setProto.set(P2, P);
+    P2.prototype = Object.create(P.prototype);
```

Comment:
Better use `$.create` instead of `Object.create`. Old browsers with `__proto__` and alternative `Promise` polyfill case.